include/aspect/stokes_matrix_free.h

```diff
@@ -443,11 +427,12 @@ namespace aspect
       MGConstrainedDoFs mg_constrained_dofs_Schur_complement;
       MGConstrainedDoFs mg_constrained_dofs_projection;
 
-      dealii::LinearAlgebra::distributed::Vector<double> active_coef_dof_vec;
-      MGLevelObject<dealii::LinearAlgebra::distributed::Vector<double> > level_coef_dof_vec;
-
       MGTransferMatrixFree<dim,double> mg_transfer_A_block;
       MGTransferMatrixFree<dim,double> mg_transfer_Schur_complement;
+
+      Table<1, VectorizedArray<double>> active_viscosity_table;
+      MGLevelObject<Table<1, VectorizedArray<double>>> level_viscosity_tables;
+      MGLevelObject<dealii::LinearAlgebra::distributed::Vector<double> > level_coef_dof_vec;
```

Comment:
Do you need to keep both (table and vector) around? Maybe leave a comment here why?

Reply:
Yes, I need to keep the level viscosities around because I use them in both the `evaluate_material_model()` (storing the matrix-free table) and `build_preconditioner()` (when I have to compute the diagonal for the shell meshes with tangential boundary) functions.

Another option is to recompute the active and level viscosities again at the beginning of `build_preconditioner()`. We would save memory at the cost of an extra loop over cells (to project viscosity to DGQ0 or DGQ1) and an MGTransfer. 

I'll add a comment now, but if you don't think the extra computation is too expensive, I think the other way is a ok too.

Reply:
I let you decide what makes more sense. Keeping the vector around during setup and then deleting it is probably a decent approach?

Reply:
Ok, we'll keep it as is.